server/test/suites/categoryRouter.test.js

```diff
@@ -0,0 +1,169 @@
+
+// Set up Chai
+const expect = chai.expect;
+chai.use(chaiHttp);
+
+// Mock documents used for testing
+let item1;
+let item2;
+let testUser1;
+let testUser2;
+let testCategory1;
+let testCategory2;
+let testCategory3;
+
+module.exports = function categorySuite() {
+  before(async () => {
+    // Connect to in-memory mongodb
+    mongoServer = new MongoMemoryServer();
+    const mongoUri = await mongoServer.getUri();
+    getMongoDB(mongoUri);
+
+    // Prepare mock documents
+    testUser1 = new User({ email: 'lobster1@gmile.com' });
+    testUser2 = new User({ email: 'lobster2@gmile.com' });
+
+    testCategory1 = new Category({
+      title: 'User1 Category1',
+      userId: testUser1._id,
+    });
+
+    testCategory2 = new Category({
+      title: 'User2 Category1',
+      userId: testUser2._id,
+    });
+
+    testCategory3 = new Category({
+      title: 'User2 Category2',
+      userId: testUser2._id,
+    });
+
+    item1 = new Item({
+      sourceLink: 'www.googe.com',
+      placesId: 'something',
+      userId: testUser1._id,
+      highlight: 'highlight words',
+      comment: 'sample comments',
+      categoryIds: [testCategory1._id],
+    });
+
+    item2 = new Item({
+      sourceLink: 'www.googe.com',
+      placesId: 'something else',
+      userId: testUser1._id,
+      highlight: 'other highlighted words',
+      comment: 'another sample comment',
+      categoryIds: [testCategory1._id],
+    });
+
+    testCategory1.items = [item1._id, item2._id];
+  });
+
+  after(async () => {
+    await mongoose.disconnect();
+    await mongoServer.stop();
+  });
+
+  /**
+   * Category Router Tests
+   */
+  describe('/createCategory', function () {
+    after(async () => {
+      await Category.remove({});
+    });
+
+    it('add a new category by user2', (done) => {
+      chai
+        .request(server)
+        .post('/category/createCategory')
+        .send({
+          userId: testUser2._id,
+          title: 'User2 New Category',
+        })
+        .set('content-type', 'application/json')
+        .end(async function (error, response) {
+          expect(response).to.have.status(200);
+          try {
+            const categories = await Category.find({});
+            expect(categories).to.have.lengthOf(1);
+            expect(categories[0].title).equals('User2 New Category');
+            done();
+          } catch (error) {
+            expect(error).to.be.null;
+          }
+        });
+    });
+  });
+
+  describe('/getCategories', () => {
+    before(async () => {
+      await testCategory1.save();
+      await testCategory2.save();
+      await testCategory3.save();
+    });
+    after(async () => {
+      await Category.remove({});
+    });
+
+    it('error status and no response when no userId is provided in the request', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategories')
+        .end((error, response) => {
+          expect(response).to.not.have.status(200);
```

Comment:
I think it would prob be better to expect whatever the error code actually will be instead of expecting that it won't be 200.

Reply:
asked offline